### ISSUE TITLE
Add support for ALTER TABLE .. SET NOT NULL in Oracle

### ISSUE THREAD

**Issue by nielsmeijer** [jOOQ/jOOQ#10753]
When changing the nullability of a column in a table using jOOQ on Oracle dialect, an oracle exception is raised: ```ORA-00902: invalid datatype```.

## Code sample:

```java
DSL.using(SQLDialect.ORACLE12C)
.alterTable(name("schema_name", "table_name"))
.alterColumn(field(name("field_name")))
.setNotNull();
.execute();
```
jOOQ translates this code the following SQL query

```sql
alter table "schema_name"."table_name"
  modify "field_name" set not null
```

The query fails on the database with ORA error ```ORA-00902```

A correct query for changing the nullability looks as followed:

```sql
alter table "schema_name"."table_name"
  modify ("field_name" not null)
```

The same issue applies for dropping not null contraints:

```java
DSL.using(SQLDialect.ORACLE12C)
.alterTable(name("schema_name", "table_name"))
.alterColumn(field(name("field_name")))
.dropNotNull();
.execute();
```

### Versions:

- jOOQ: tested on ```3.13.2 (pro)``` and ```3.13.5 (pro)```
- Database: ```Oracle Database 12c Enterprise Edition Release 12.2.0.1.0 ```
- JDBC Driver: oracle ojdbc7

**Comment by lukaseder** [comment]
Thank you very much for your message.

The `SET NOT NULL` clause of the `ALTER TABLE` statement is currently not supported in Oracle, see `@Support` annotation on [`AlterTableAlterStep::setNotNull`](https://www.jooq.org/javadoc/latest/org.jooq/org/jooq/AlterTableAlterStep.html#setNotNull()), which includes:

```java
@Support({ASE,AURORA_POSTGRES,COCKROACHDB,DB2,DERBY,FIREBIRD_3_0,H2,HSQLDB,POSTGRES,VERTICA})
```

We could support it for the next minor release via

```sql
ALTER TABLE MODIFY <col> [ NOT ] NULL;
```

I'll verify why it hasn't been included yet.
```

**Comment by lukaseder** [comment]
I'm not sure why it hasn't been included yet. There doesn't seem to be any technical reason. We'll include this in 3.14.0 (just in time!), and backport to 3.13.6 (#10754).

Thanks again for your report.

**Comment by lukaseder** [comment]
Fixed in 3.14.0 and 3.13.6 (#10754). The fix will be included in the next nightly builds (available here: https://www.jooq.org/download/versions), but we'll also soon release the two versions.

**Comment by nielsmeijer** [comment]
Thanks for the quick fix!
